libraries/node-core-library/src/JsonFile.ts

```diff
@@ -111,6 +111,13 @@ export class JsonFile {
     }
   }
 
+  /**
+   * Parses a JSON file's contents.
+   */
+  public static parse(jsonContents: string): JsonObject {
```

Comment:
Should we call it `JsonFile.parseString()`?  When I see `JsonFile.parse()` it feels like the argument should be a filename.

In retrospect perhaps this class would have been better named `Json` instead of `JsonSchema`.

Reply:
Suppose (hypothetically) that `JsonFile.load('folder/input.json')` would have printed this error message:

```
Error: Unexpected character "[" at ./folder/inputs/test.json:15:3
```

It may be the case that `JsonFile.parse(content)` is parsing `content` which was read from `folder/input.json`. Often it is useful to pass that information to enable a more accurate error message.  We use option called `filenameForErrors` for this.  Currently there are two APIs with this feature:

- [JsonSchema.validateObject()](https://github.com/microsoft/rushstack/blob/5627415032539c27e93fdddac4edf93d2c87a311/libraries/node-core-library/src/JsonSchema.ts#L253)
- [ExtractorConfig.prepare()](https://github.com/microsoft/rushstack/blob/aac158f7d662b76fd2b913400bc0f8e252c2c647/apps/api-extractor/src/api/ExtractorConfig.ts#L450)

I wonder if it would make sense here as well. 

For `JsonSchema.validateObject()` I thought it was important enough information that I made it a required parameter (and you must explicitly opt-out by passing an empty string if you don't have a filename). That signature might be confusing here though, since two `string` adjacent parameters can get confused easily.

Reply:
`updateString()` uses this naming pattern. And its parameter is `previousJson`, so maybe `jsonContents` should be just `json`?

Reply:
I'm good with `parseString